spikeinterface/toolkit/postprocessing/template_metrics.py

```diff
@@ -23,37 +24,100 @@ def get_template_metric_names():
-    
+    Compute template metrics including:
+    * peak_to_valley
+    * peak_trough_ratio
+    * halfwidth
+    * repolarization_slope
+    * recovery_slope
+
+    Parameters
+    ----------
+    waveform_extractor : WaveformExtractor, optional
+        The waveform extractor used to compute template metrics
+    templates : np.array, optional
+        If templates is specified and waveform_extractor is None, metrics are computed on
+        the provided template -- shape: ()
+    feature_names : list, optional
+        List of metrics to compute (see si.toolkit.get_template_metric_names()), by default None
+    peak_sign : str, optional
+        "pos" | "neg", by default 'neg'
+    upsampling_factor : int, optional
+        Upsample factor, by default 10
+    sparsity_dict: dict or None
+        Default is sparsity_dict=None and template metric is computed on extremum channel only.
+        If given, the dictionary should contain a sparsity method (e.g. "best_channels") and optionally
+        arguments associated with the method (e.g. "num_channels" for "best_channels" method).
+        Other examples are:
+           * by radius: sparsity_dict=dict(method="radius", radius_um=100)
+           * by SNR threshold: sparsity_dict=dict(method="threshold", threshold=2)
+           * by property: sparsity_dict=dict(method="by_property", by_property="group")
+        For more info, see the toolkit.get_template_channel_sparsity() function.
+    **kwargs: keyword arguments for get_recovery_slope function:
+        * window_ms: window in ms after the positiv peak to compute slope
+
+    Returns
+    -------
+    tempalte_metrics : pd.DataFrame
+        Dataframe with the computed template metrics. 
+        If 'sparsity_dict' is None, the index is the unit_id.
+        If 'sparsity_dict' is given, the index is "unit_id@channel_id"
     """
     unit_ids = waveform_extractor.sorting.unit_ids
     sampling_frequency = waveform_extractor.recording.get_sampling_frequency()
 
     if feature_names is None:
         feature_names = list(_metric_name_to_func.keys())
 
-    extremum_channels_inds = get_template_extremum_channel(waveform_extractor, peak_sign=peak_sign,
-                                                           outputs='index')
+    if sparsity_dict is None:
+        extremum_channels_ids = get_template_extremum_channel(waveform_extractor, peak_sign=peak_sign,
+                                                              outputs='id')
 
-    template_metrics = pd.DataFrame(index=unit_ids, columns=feature_names)
+        template_metrics = pd.DataFrame(
+            index=unit_ids, columns=[feature_names])
+    else:
+        extremum_channels_ids = get_template_channel_sparsity(
+            waveform_extractor, **sparsity_dict)
+        df_indices = []
+        for unit_id, sparse_channels in extremum_channels_ids.items():
+            new_indices = [f"{unit_id}@{chan}" for chan in sparse_channels]
```

Comment:
You should use the pandas multi index for that instead.

Reply:
ahhh will look into it!

Reply:
done